src/main/java/org/elasticsearch/index/mapper/core/DateFieldMapper.java

```diff
@@ -253,6 +258,15 @@ public void setTimeUnit(TimeUnit timeUnit) {
             this.dateMathParser = new DateMathParser(dateTimeFormatter);
         }
 
+        public boolean isIndexCreatedBefore2x() {
```

Comment:
Just to get this right: You have a `FallbackDateFormatter` that has support for this? My approach was to put this into the mapper as this is the component that acts different on the version, but no string feelings here.